src/core/io/writer/h5md/h5md_core.cpp

```
@@ -257,73 +257,90 @@ void File::Close()
+                f[0][particle_index][0] = current_particle->f.f[0] * fac;
+                f[0][particle_index][1] = current_particle->f.f[1] * fac;
+                f[0][particle_index][2] = current_particle->f.f[2] * fac;
+            }
+
+}
+
+void File::Write(int write_dat)
+{
+#ifdef H5MD_DEBUG
+    std::cout << "Called " << __func__ << " on node " << this_node << std::endl;
+#endif
+    bool write_typ = write_dat & W_TYPE;
+    bool write_pos = write_dat & W_POS;
+    bool write_vel = write_dat & W_V;
+    bool write_force = write_dat & W_F;
+    bool write_mass = write_dat & W_MASS;
+    int num_particles_to_be_written;
+    if(m_write_ordered==true && n_nodes==1)
+        num_particles_to_be_written=n_part;
+    else
+        num_particles_to_be_written=cells_get_n_particles();
+
+    double_array_3d pos(boost::extents[1][num_particles_to_be_written][3]);
+    double_array_3d vel(boost::extents[1][num_particles_to_be_written][3]);
+    double_array_3d f(boost::extents[1][num_particles_to_be_written][3]);
+    int_array_3d image(boost::extents[1][num_particles_to_be_written][3]);
+    int_array_3d id(boost::extents[1][num_particles_to_be_written][1]);
+    int_array_3d typ(boost::extents[1][num_particles_to_be_written][1]);
+    double_array_3d mass(boost::extents[1][num_particles_to_be_written][1]);
+
+    if (m_write_ordered==true && n_nodes==1) {
+        
+        //loop over all particles
+        for(int particle_index=0;particle_index<n_part;particle_index++){
+            Particle current_particle;
+	        get_particle_data(particle_index, &current_particle); //this function only works when run with one process
+            fill_arrays_for_h5md_write_with_particle_property(particle_index, id, typ, mass, pos, image, vel, f, &current_particle,  write_typ, write_mass, write_pos, write_vel, write_force );
+	        free_particle(&current_particle);
+        }
+
+    }else{
```

if m_write_ordered == true && n_nodes != 1 this should throw an exception.